File: app/src/main/java/com/persona5dex/repositories/MainPersonaRepository.java
package com.persona5dex.repositories;

import androidx.lifecycle.LiveData;

import com.persona5dex.models.MainListPersona;

import java.util.List;

/**
 * Created by Rechee on 11/18/2017.
 */

public interface MainPersonaRepository {
    LiveData<List<MainListPersona>> getAllPersonasForMainListLiveData();
    List<MainListPersona> getAllPersonasForMainList();
    LiveData<List<MainListPersona>> getDLCPersonas();
    LiveData<String> getPersonaName(int personaID);
}

File: app/src/main/java/com/persona5dex/repositories/MainPersonaRepository.kt
package com.persona5dex.repositories

import androidx.lifecycle.LiveData
import com.persona5dex.models.MainListPersona

/**
 * Created by Rechee on 11/18/2017.
 */
interface MainPersonaRepository {
    val allPersonasForMainListLiveData: LiveData<List<MainListPersona>>
    val allPersonasForMainList: List<MainListPersona>
    val dLCPersonas: LiveData<List<MainListPersona>>
    fun getPersonaName(personaID: Int): LiveData<String?>
}